src/client/js/dialog-handler.js

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const triggerBtns = document.querySelectorAll('button[data-dialog]')
+let contentMap, dialogEl // expect a single dialog on page at any time
+
+function init () {
+  contentMap = new Map()
+  dialogEl = document.createElement('dialog')
+  document.body.append(dialogEl)
+
+  triggerBtns.forEach(btn => btn.addEventListener('click', handleEvent))
+}
+
+function handleEvent (e) {
+  switch (true) {
+    case e.target.matches('button[data-dialog]'):
+      openDialog(e.target.dataset.dialog)
+      break
+    case e.target.matches('dialog button.close'):
+    case e.target === dialogEl: // ::backdrop click
+      closeDialog()
+      break
+  }
+}
+
+async function openDialog (partialName) {
+  dialogEl.showModal() // provide immediate UI response by showing ::backdrop regardless of dialog content
+  dialogEl.addEventListener('click', handleEvent)
+
+  const content = contentMap.get(partialName) ?? await fetchContent(partialName)
```

Comment:
Question (non-blocking): Let’s assume the user is on a very slow or flacky connection they just would not get any form of feedback — does a loading state for dynamic content would make sense here?

Reply:
Yes, a loading state for dynamic content makes sense.  We can/should add that – I'm not sure if we have any loading animation assets from UX yet.  As you mention, this would only be visible for very slow connections, since we're displaying the HTML immediately – images and other media assets would load afterwords.

That being said, the current logic _should_ already be built to provide immediate feedback as mentioned in the comment.  (Let me know if that comment isn't clear enough, or maybe I don't understand.)  As currently programmed, the backdrop (an 80% opacity black scrim) will appear immediately regardless of available content.  If/when content is loaded, that content will fade in over the backdrop.  By "loaded", I mean just the HTML, which should be fast.  Images would load subsequently.  If content fails to load, the scrim is expected to disappear to avoid blocking the user.

Reply:
That comment was really just directed toward dynamic content within the modal — thanks for elaborating. In that case we should be good.